pytorch_lightning/core/lightning.py

```diff
@@ -428,6 +429,29 @@ def log(
                 "With `def training_step(self, dataloader_iter)`, `self.log(..., batch_size=...)` should be provided."
             )
 
+        reduce_fx = reduce_fx.lower() if isinstance(reduce_fx, str) else reduce_fx
+
+        # check if we have extracted the batch size already
```

Comment:
Can we move this logic within the ResultCollection instead?

Reply:
I think the ResultCollection should keep track of the batch and recompute the batch_size only if the batch changed and log is been performed.

Reply:
done! result collection keeps track of the batch, and update & stores the batch size whenever it's required.